Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/Bookmarks/AddEditBookmarkTableViewController.swift

```diff
@@ -360,8 +360,22 @@ class AddEditBookmarkTableViewController: UITableViewController {
         isLoading = true
         
         for tab in tabs {
-            if let url = tab.url, url.isWebPage(), !url.isAboutHomeURL {
-                Bookmarkv2.add(url: url, title: tab.title, parentFolder: parentFolder)
+            if PrivateBrowsingManager.shared.isPrivateBrowsing {
+                if let url = tab.url, url.isWebPage(), !url.isAboutHomeURL {
+                    Bookmarkv2.add(url: url, title: tab.title, parentFolder: parentFolder)
+                }
+            } else {
+                if let tabID = tab.id {
+                    TabMO.fetchManagedTabObject(tabID: tabID) { fetchedTab in
+                        if let urlString = fetchedTab?.url, let url = URL(string: urlString), url.isWebPage(), !url.isAboutHomeURL {
+                            DispatchQueue.main.async {
```

Comment:
if we fetch on main thread this dispatch can be removed

Reply:
Fix at https://github.com/brave/brave-ios/pull/4133/commits/6745aeefbbe54dffe83e8f6de4b6732530eddb7d